utils/soca/gdas_soca_diagb.h

```diff
@@ -429,21 +430,28 @@ namespace gdasapp {
       /// Impose an exponential decay to the background error
       // ----------------------------------------------------
       if (fullConfig.has("vertical e-folding scale")) {
-        double efold = 0.0;
-        fullConfig.get("vertical e-folding scale", efold);
+        double efold = fullConfig.getDouble("vertical e-folding scale", 500.0);
+        double edRatio = fullConfig.getDouble("min efold depth ratio", 3.0);
+        double localEfold = 0.0;
         for (auto & var : configD.socaVars.variables()) {
-        oops::Log::info()
+          oops::Log::info()
            << "====================== apply exponential decay to the background error. "
            << " e-folding scale: " << efold << " m for " << var << std::endl;
           auto stdDevBkg = atlas::array::make_view<double, 2>(bkgErrFs[var]);
           auto numLevels = xbFs["sea_water_potential_temperature"].shape(0);
           for (atlas::idx_t jnode = 0; jnode < numLevels; ++jnode) {
+            if (ghostView(jnode) > 0) {
+              continue;  // Skip ghost cells
+            }
             for (atlas::idx_t level = 0; level < xbFs[var].shape(1); ++level) {
-              if (viewDepth(jnode, level) >= 0.0) {
-                stdDevBkg(jnode, level) *= std::exp(-viewDepth(jnode, level) / efold);
+              if (viewBathy(jnode, 0) > 0.0) {
```

Comment:
should this be comparing to `depthMin`?

Reply:
Oh, doesn't matter then.

Reply:
That was my main concern in this PR, and that's because I didn't understand that depthMin is addressed differently.